collector/application/application.go

```diff
@@ -106,17 +108,20 @@ func (a *Application) buildPipeline() error {
 	// 1. Kindling Metric Format Processor
 	formatProcessorFactory := a.componentsFactory.Processors[kindlingformatprocessor.ProcessorName]
 	formatProcessor := formatProcessorFactory.NewFunc(formatProcessorFactory.Config, a.telemetry.Telemetry, otelExporter)
+	aggregateProcessorFactory := a.componentsFactory.Processors[aggregateprocessor.Type]
+	aggregateProcessor := aggregateProcessorFactory.NewFunc(aggregateProcessorFactory.Config, a.telemetry.Telemetry, formatProcessor)
 	// 2. Kubernetes metadata processor
 	k8sProcessorFactory := a.componentsFactory.Processors[k8sprocessor.K8sMetadata]
-	k8sMetadataProcessor := k8sProcessorFactory.NewFunc(k8sProcessorFactory.Config, a.telemetry.Telemetry, formatProcessor)
+	k8sMetadataProcessor := k8sProcessorFactory.NewFunc(k8sProcessorFactory.Config, a.telemetry.Telemetry, aggregateProcessor)
 	// Initialize all analyzers
 	// 1. Common network request analyzer
 	networkAnalyzerFactory := a.componentsFactory.Analyzers[network.Network.String()]
 	// Now NetworkAnalyzer must be initialized before any other analyzers, because it will
 	// use its configuration to initialize the conntracker module which is also used by others.
 	networkAnalyzer := networkAnalyzerFactory.NewFunc(networkAnalyzerFactory.Config, a.telemetry.Telemetry, []consumer.Consumer{k8sMetadataProcessor})
 	// 2. Layer 4 TCP events analyzer
-	k8sMetadataProcessor2 := k8sProcessorFactory.NewFunc(k8sProcessorFactory.Config, a.telemetry.Telemetry, otelExporter)
+	aggregateProcessorForTcp := aggregateProcessorFactory.NewFunc(aggregateProcessorFactory.Config, a.telemetry.Telemetry, otelExporter)
```

Comment:
Due to different subsequent processing components, multiple instances need to be created; The same problem will also occur on conntrack (if we later treat it as a separate processor). I think this problem can be avoided if the information of subsequent nodes of the component is maintained by an external manager

Reply:
I'm not sure whether I got your point correctly. I think multiple instances won't bring many problems. Why do we need an external manager?

Reply:
Processors must be able to be created multiple times as we will make it possible that the pipeline could be configured later. For those components that don't support multiple instances, refactoring will be conducted.